sdk/ai/azure-ai-agents/samples/agents_tools/sample_agents_computer_use.py

```diff
@@ -0,0 +1,205 @@
+    agent = agents_client.create_agent(
+        model=os.environ["MODEL_DEPLOYMENT_NAME"],
+        name="my-agent-computer-use",
+        instructions="""
+            You are an computer automation assistant. 
+            Use the computer_use_preview tool to interact with the screen when needed.
+            """,
+        tools=computer_use.definitions,
+    )
+
+
+    print(f"Created agent, ID: {agent.id}")
+
+    # Create thread for communication
+    thread = agents_client.threads.create()
+    print(f"Created thread, ID: {thread.id}")
+
+    input_message = ("I can see a web browser with bing.com open and the cursor in the search box."
+                     "Type 'movies near me' without pressing Enter or any other key. Only type 'movies near me'.")
+    image_base64 = image_to_base64(asset_file_path)
+    img_url = f"data:image/jpeg;base64,{image_base64}"
+    url_param = MessageImageUrlParam(url=img_url, detail="high")
+    content_blocks: List[MessageInputContentBlock] = [
+        MessageInputTextBlock(text=input_message),
+        MessageInputImageUrlBlock(image_url=url_param),
+    ]
+    # Create message to thread
+    message = agents_client.messages.create(
+        thread_id=thread.id,
+        role=MessageRole.USER,
+        content=content_blocks
+    )
+    print(f"Created message, ID: {message.id}")
+
+    run = agents_client.runs.create(thread_id=thread.id, agent_id=agent.id)
+    print(f"Created run, ID: {run.id}")
+
+    #create a fake screenshot showing the text typed in
+    result_image_base64 = image_to_base64(action_result_file_path)
+    result_img_url = f"data:image/jpeg;base64,{result_image_base64}"
+    computer_screenshot = ComputerScreenshot(image_url=result_img_url)
+
+    while run.status in ["queued", "in_progress", "requires_action"]:
+        time.sleep(1)
+        run = agents_client.runs.get(thread_id=thread.id, run_id=run.id)
+
+        if run.status == "requires_action" and isinstance(run.required_action, SubmitToolOutputsAction):
+            print("Run requires action:")
+            tool_calls = run.required_action.submit_tool_outputs.tool_calls
+            if not tool_calls:
+                print("No tool calls provided - cancelling run")
+                agents_client.runs.cancel(thread_id=thread.id, run_id=run.id)
+                break
+
+            tool_outputs = []
+            for tool_call in tool_calls:
+                if isinstance(tool_call, RequiredComputerUseToolCall):
+                    print(tool_call)
+                    try:
+                        action = tool_call.computer_use_preview.action
+                        print(f"Executing computer use action: {action.type}")
+                        if isinstance(action, TypeAction):
+                            print(f"  Text to type: {action.text}")
+                            #add hook to input text in managed environment API here
+                            
```

Comment:
Should this comment be here? `#add hook to input text in managed environment API here`

Reply:
Yes, this example is somewhat artificial, demonstrating how to receive an action and send back output in this scenario. A real implementation would be hooked into a managed environment framework like Playwright or Docker and map the action types to specific API calls to perform said action.

Reply:
Oh, I see you are hinting to the customer that this is the location where they should provide a real screenshot. I would combine this comment with the one below, to say "# Add a fake .... . Application will provide a true screenshot of ... ".